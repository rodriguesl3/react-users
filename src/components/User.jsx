import React from 'react';

const User = ({ user }) => {
  return <div>this is a test => {user.name}</div>;
};

export default User;
