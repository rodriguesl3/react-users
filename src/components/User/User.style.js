import styled from 'styled-components';

export const UserCard = styled.div`
  display: flex;
  flex-direction: column;
  width: 45%;
  margin: 10px;
  border: 2px solid black;
  border-radius: 25px;
  justify-content: center;
  align-items: center;
  padding-top: 20px;
  padding-bottom: 20px;

  .company-section {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;
  }
`;

export const Divider = styled.hr`
  width: 80%;
`;
