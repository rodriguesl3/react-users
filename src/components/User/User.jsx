import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FaArrowDown } from 'react-icons/fa';

import { UserCard, Divider } from './User.style';

const User = ({ user }) => {
  const [showCatchPhrase, setShowCatchPhrase] = useState(false);

  return user ? (
    <UserCard>
      <h2>General Information</h2>
      <div>Name: {user.name}</div>
      <div>Email: {user.email}</div>
      <Divider />
      <h2>Address</h2>
      <div>Street: {`${user.address.street}, ${user.address.suite || ''}`}</div>
      <div>City: {user.address.city}</div>
      <div>Zip Code: {user.address.zipcode || ''}</div>
      <Divider />
      <h2>WebSite</h2>
      <div>{user.website}</div>
      <Divider />

      <div
        className="company-section"
        onMouseEnter={() => setShowCatchPhrase(true)}
        onMouseLeave={() => setShowCatchPhrase(false)}
      >
        <h2>Company</h2>
        <div>{user.company.name}</div>
        <Divider />
        <small>View More...</small>
        <FaArrowDown />
        {showCatchPhrase && user.company.catchPhrase}
      </div>
    </UserCard>
  ) : (
    <div>User Not Found</div>
  );
};

User.prototype = {
  user: PropTypes.object.isRequired,
};

export default User;
