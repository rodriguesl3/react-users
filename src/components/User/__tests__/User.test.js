import React from 'react';
import { shallow } from 'enzyme';
import User from '../User';

describe('Link', () => {
  it('Renders User', () => {
    const userMock = {
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz',
      address: {
        street: 'Kulas Light',
        suite: 'Apt. 556',
        city: 'Gwenborough',
        zipcode: '92998-3874',
        geo: {
          lat: '-37.3159',
          lng: '81.1496',
        },
      },
      phone: '1-770-736-8031 x56442',
      website: 'hildegard.org',
      company: {
        name: 'Romaguera-Crona',
        catchPhrase: 'Multi-layered client-server neural-net',
        bs: 'harness real-time e-markets',
      },
    };

    const userComponent = shallow(<User user={userMock} />);
    expect(userComponent).toMatchSnapshot();
  });

  //TODO: app
  it('Renders User Not Found', () => {
    const userComponent = shallow(<User />);
    const expectResponse = 'User Not Found';
    const text = userComponent.text();
    console.log(text);
    expect(expectResponse).toEqual(text);
    expect(userComponent).toMatchSnapshot();
  });
});
