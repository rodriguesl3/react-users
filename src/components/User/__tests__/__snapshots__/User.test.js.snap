// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Renders User 1`] = `ShallowWrapper {}`;

exports[`Link Renders User 2 1`] = `ShallowWrapper {}`;

exports[`Link Renders User Not Found 1`] = `ShallowWrapper {}`;
